python/ray/tune/checkpoint_manager.py

```diff
@@ -112,7 +112,8 @@ def on_checkpoint(self, checkpoint):
         self.newest_persistent_checkpoint = checkpoint
 
         # Remove the old checkpoint if it isn't one of the best ones.
-        if old_checkpoint.value and old_checkpoint not in self._membership:
+        if old_checkpoint.value and old_checkpoint not in self._membership \
+           and old_checkpoint.value != checkpoint.value:
```

Comment:
@krfricke is it possible to add a comprehensive comment (as you already did on the PR description) into the code? Just to explain what you're doing.

Reply:
Also, is there an invariant here? (i.e., don't process the checkpoint if the checkpoint is already tracked?)